### PR TITLE
add functions for pre-committing, committing, staging data, and unsealing

### PR DESCRIPTION
**Pull request by laser**
## Why does this PR exist?

Lotus wants to seal w/out having to use a sector builder.

## What's in this PR?

This PR exposes primitive proof operations (write a piece, pre-commit, commit, unseal) to library consumers. This will allow them to build their own sector builder-like things.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,6 +22,12 @@ libc = "0.2.58"
 log = "0.4.7"
 pretty_env_logger = "0.3.0"
 once_cell = "0.2.4"
+nodrop = "0.1.14"
```
Comment (dignifiedquire) [inline Cargo.toml]: from the crate docs of nodrop

> Deprecated: Use ManuallyDrop or MaybeUninit instead!

Comment (dignifiedquire) [reply Cargo.toml]: https://docs.rs/nodrop/0.1.14/nodrop/

File: src/api.rs [hunk 2]
```diff
@@ -1,33 +1,350 @@
...
+use storage_proofs::hasher::pedersen::PedersenDomain;
 
-#[repr(C)]
-#[derive(Clone)]
-pub struct FFIPublicPieceInfo {
-    pub num_bytes: u64,
-    pub comm_p: [u8; 32],
+lazy_static! {
+    static ref TEMPORAY_AUX_MAP: Mutex<HashMap<u64, api_types::TemporaryAux>> =
+        { Mutex::new(HashMap::new()) };
```
Comment (dignifiedquire) [inline src/api.rs]: no need for those curly braces

File: Cargo.toml (hunk 1 above)
Comment (dignifiedquire) [reply Cargo.toml]: https://doc.rust-lang.org/stable/std/mem/struct.ManuallyDrop.html

I guess I missed that the first time around

File: Cargo.toml [hunk 3]
```diff
@@ -22,6 +22,12 @@ libc = "0.2.58"
...
+lazy_static = "1.4.0"
```
Comment (dignifiedquire) [inline Cargo.toml]: I believe you can use `once_cell` instead, which is already a dependency

File: src/api.rs (hunk 2 above)
Comment (laser) [reply src/api.rs]: fixed

File: Cargo.toml (hunk 1 above)
Comment (laser) [reply Cargo.toml]: fixed

File: Cargo.toml (hunk 3 above)
Comment (laser) [reply Cargo.toml]: fixed

**Pull request merged**